File: web/src/model/record.js
import { Money } from "./money";

export function parse({ id, uuid, products, discounts, price, time, info }) {
  return {
    name: "record",
    id,
    uuid,
    products,
    discounts,
    price: Money.fromJSON(price),
    time: new Date(time),
    info,
  };
}

File: android/app/src/main/kotlin/com/cameldridge/conartist/model/Record.kt
package com.cameldridge.conartist.model

import android.os.Parcelable
import com.cameldridge.conartist.services.api.graphql.fragment.RecordFragment
import kotlinx.android.parcel.Parcelize
import java.util.Date
import java.util.UUID

@Parcelize
data class Record(
  val id: Id,
  val products: List<Id>,
  val price: Money,
  val info: String,
  val uuid: UUID?,
  val time: Date
): Parcelable {
  companion object {
    fun fromFragment(fragment: RecordFragment) = Record(
      Id.Id(fragment.id),
      fragment.products.map { Id.Id(it) },
      fragment.price,
      fragment.info,
      fragment.uuid,
      fragment.time
    )
  }
}
